### PR TITLE
[risk=low][RW-11368] Upgrade Java to 17

### PR DESCRIPTION
**Pull request by jmthibault79**
Tested by deploying a java 17 runtime to GAE in the Test environment and observing no problems when routing traffic to it.

If you want to deploy a version yourself, you can run:
```
deploy/project.rb deploy \
--project all-of-us-workbench-test \
--account circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com \
--no-promote \
--git-version joel/java17 \
--app-version <a version ID you choose>
```

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: deploy/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 
 RUN apt-get update \
   && apt-get install --no-install-recommends -yq \
-  gettext \
-  ruby \
   default-mysql-client \
-  python3 \
-  python3-pip \
+  gettext \
+  git \
```
Comment (jmthibault79) [inline deploy/Dockerfile:10]: Added `git` and `unzip` which were missing in the new image, and sorted by name

File: deploy/Dockerfile [hunk 2]
```diff
@@ -23,6 +25,7 @@ RUN n 18
 
 RUN npm install @openapitools/openapi-generator-cli -g
 
+# keep in sync with api/gradle.build and ci/Dockerfile.circle_build
 ENV CLOUD_SDK_VERSION 392.0.0
```
Comment (jmthibault79) [inline deploy/Dockerfile:29]: I'm planning to rev this too, but in a different PR

File: ci/Dockerfile.circle_build [hunk 3]
```diff
@@ -28,8 +29,15 @@ RUN cd && \
   ~/google-cloud-sdk/bin/gcloud components install app-engine-java && \
   rm -rf gcloud.tgz
 
-RUN sudo apt-get update
-RUN sudo apt-get install gettext ruby default-mysql-client python3-pip wait-for-it
+RUN sudo apt-get update && sudo apt-get install --no-install-recommends -yq \
```
Comment (jmthibault79) [inline ci/Dockerfile.circle_build:32]: added cleanup steps from the other Dockerfile, for a smaller image

File: deploy/Dockerfile [hunk 4]
```diff
@@ -1,17 +1,19 @@
 # Image used for deployment. The Circle image is not used here as it does not
 # have an ARM build (M1 chip requires emulation).
-# Keep synchronized with ci/Dockerfile.circle_build
-FROM openjdk:11
+# Keep synchronized with ci/Dockerfile.circle_build and api/gradle.build
+FROM eclipse-temurin:17
```
Comment (Qi77Qi) [inline deploy/Dockerfile:4]: doesn't need to be this PR...wonder if we should use appsec approved images like these https://github.com/DataBiosphere/leonardo/blob/develop/Dockerfile#L18

**Review by Qi77Qi — APPROVED** [review]

File: deploy/Dockerfile (hunk 4 above)
Comment (jmthibault79) [reply deploy/Dockerfile:4]: interesting... do you have documents about these images?

File: ci/Dockerfile.circle_build [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # To build and deploy, from this directory:
 # $ docker build . -f Dockerfile.circle_build -t us.gcr.io/broad-dsp-gcr-public/workbench:buildimage-X.Y.Z .
-# Note: M1 Macs may need to add `--platform linux/x86_64/v8`
```
Comment (jmthibault79) [inline ci/Dockerfile.circle_build:3]: @Peter-Lavigne I had problems with the "/v8"

**Review by freemabd — COMMENTED** [review]
I've deployed this locally and it causes problems with our tanagra deployment since it's Java 11(get compile errors).  I'm using jenv and was able to set the shell instance java version to 11 using `jenv shell openjdk64-11.0.19`. This worked but I'm not sure if everyone on our team and workbench-core use jenv. Another potential workaround could just be to upgrade tanagra to Java 17, but need to check with that team.

File: deploy/Dockerfile (hunk 4 above)
Comment (yonghaoy) [reply deploy/Dockerfile:4]: Those are appsec "blessed" images
https://github.com/broadinstitute/dsp-appsec-blessed-images

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**